ipaclient/plugins/cert.py

```diff
@@ -105,6 +104,10 @@ def get_args(self):
             yield arg
 
     def forward(self, csr=None, **options):
+        # Deferred import, because ipaclient.csrgen is expensive to load
+        # see https://pagure.io/freeipa/issue/7484
+        from ipaclient import csrgen
+
```

Comment:
I think we could defer this even further, to the `if csr is None` suite (about 10 lines down from here).  Then the import will only be incurred when the csrgen feature is actually being used, instead of on all `ipa cert-request` invocations.

Reply:
Good catch!